### ISSUE TITLE
Multi-modal (Leuven) files

### ISSUE THREAD

**Issue by ferponcem** [hbp-brain-charting/public_analysis_code#24]
Another set of files where I would like to know what you think, this time for the multi-modal task @bthirion @man-shu 
 
* [Here](https://github.com/hbp-brain-charting/admin/blob/1522a2441165e7be2e0e1284522938de9c430e00/documentation/documentation.tex#L6889) we have the description. I added a table with the changes made to the general MRI parameters.
* [Here](https://github.com/hbp-brain-charting/public_analysis_code/blob/master/ibc_data/ibc_tasks.tsv#L79) I added it to the tasks file, I am not sure on how to include the modified parameters but the rest I think it's there.
* [File](https://github.com/hbp-brain-charting/public_analysis_code/blob/master/ibc_data/ibc_conditions.tsv#L481) with the list of conditions, let me know if it's enough.

* [File](https://github.com/hbp-brain-charting/public_analysis_code/blob/feature/contrasts_descriptions/ibc_data/contrasts/Multi-modal.csv) with the contrasts in matrix form. I copied what we have in the [contrasts script](https://github.com/hbp-brain-charting/public_analysis_code/blob/a3f11818e521f90f85c92b2cf13ec7ce5b7b28e4/ibc_public/utils_contrasts.py#L225) but I am not sure that the weights are right, because sometimes the decimal positions can be problematic.

@bthirion I also think we're missing 1 control condition on the contrasts script, because we have this `no_valves_open_tacnotac.bmp` event, see [example](https://github.com/hbp-brain-charting/analysis_pipeline/blob/master/ibc_main/neurospin_data/info/sub-04/leuven/task-leuven_sub-04_run-01_events.tsv#L92) but then I can't find it on the `tactile_` definition of the contrasts, maybe I am not seeing it?

Also, I don't think we've come to an agreement on what to do with those intermediate contrast, for example we have this `animate` contrast defined, but we just use it to then define another contrast: `animate-inanimate`, so should I also add them as columns to the file??

Thank you!

**Comment by man-shu** [comment]
Just made some minor changes to the description in this commit 4e0b2955b9c76bacd50ced022fd69d0b8b7684be for both Multi modal and MDTB

**Comment by bthirion** [comment]
> * [File](https://github.com/hbp-brain-charting/public_analysis_code/blob/feature/contrasts_descriptions/ibc_data/contrasts/Multi-modal.csv) with the contrasts in matrix form. I copied what we have in the [contrasts script](https://github.com/hbp-brain-charting/public_analysis_code/blob/a3f11818e521f90f85c92b2cf13ec7ce5b7b28e4/ibc_public/utils_contrasts.py#L225) but I am not sure that the weights are right, because sometimes the decimal positions can be problematic

You're right, this is not great. Maybe we could replace the  fractional number by integers whenever possible.
(1, 1, 1, -3) instead of (.33, .33, .33, -1)
Can you issue a PR for that on utils_contrasts ?
Thx  !

**Comment by ferponcem** [comment]
> > * [File](https://github.com/hbp-brain-charting/public_analysis_code/blob/feature/contrasts_descriptions/ibc_data/contrasts/Multi-modal.csv) with the contrasts in matrix form. I copied what we have in the [contrasts script](https://github.com/hbp-brain-charting/public_analysis_code/blob/a3f11818e521f90f85c92b2cf13ec7ce5b7b28e4/ibc_public/utils_contrasts.py#L225) but I am not sure that the weights are right, because sometimes the decimal positions can be problematic
> 
> You're right, this is not great. Maybe we could replace the fractional number by integers whenever possible. (1, 1, 1, -3) instead of (.33, .33, .33, -1) Can you issue a PR for that on utils_contrasts ? Thx !

That was the idea, the issue comes when we have uneven number of conditions we want to contrast, for example `audio-visual`, we have 6 positive conditions for *audio* and 10 negative for *visual*, so if I give a -1 to each of the negative conditions, I would still need to do 10/6 so still not an integer. I wonder how you've dealt with this in the past. Maybe just going to a higher factor, eg. having 5*(6 positive conditions) vs 3*(10 negative conditions)?

**Comment by bthirion** [comment]
> * [Here](https://github.com/hbp-brain-charting/public_analysis_code/blob/master/ibc_data/ibc_tasks.tsv#L79) I added it to the tasks file, I am not sure on how to include the modified parameters but the rest I think it's there.

l.6978 engagged -> engaged

Another point: I see that you prettify the contrast names, e.g. by using ' -- ' instead of '-'. I think  that this is not a good thing and that we should used the programmatically defined contrasts ids. However, we have to be consistent across tasks, hence if it has been done in the same way for other tasks, then we should leave it as is for the moment, and address this later.

**Comment by bthirion** [comment]
> * [File](https://github.com/hbp-brain-charting/public_analysis_code/blob/feature/contrasts_descriptions/ibc_data/contrasts/Multi-modal.csv) with the contrasts in matrix form. I copied what we have in the [contrasts script](https://github.com/hbp-brain-charting/public_analysis_code/blob/a3f11818e521f90f85c92b2cf13ec7ce5b7b28e4/ibc_public/utils_contrasts.py#L225) but I am not sure that the weights are right, because sometimes the decimal positions can be problematic.

Can you address my comment above ?

**Comment by ferponcem** [comment]
> > * [File](https://github.com/hbp-brain-charting/public_analysis_code/blob/feature/contrasts_descriptions/ibc_data/contrasts/Multi-modal.csv) with the contrasts in matrix form. I copied what we have in the [contrasts script](https://github.com/hbp-brain-charting/public_analysis_code/blob/a3f11818e521f90f85c92b2cf13ec7ce5b7b28e4/ibc_public/utils_contrasts.py#L225) but I am not sure that the weights are right, because sometimes the decimal positions can be problematic.
> 
> Can you address my comment above ?

Yes, that's what I am trying to address [in this comment](https://github.com/individual-brain-charting/public_analysis_code/issues/24#issuecomment-1488344508) , I am not quite sure for this particular example

**Comment by ferponcem** [comment]
> > * [Here](https://github.com/hbp-brain-charting/public_analysis_code/blob/master/ibc_data/ibc_tasks.tsv#L79) I added it to the tasks file, I am not sure on how to include the modified parameters but the rest I think it's there.
> 
> l.6978 engagged -> engaged
> 
> Another point: I see that you prettify the contrast names, e.g. by using ' -- ' instead of '-'. I think that this is not a good thing and that we should used the programmatically defined contrasts ids. However, we have to be consistent across tasks, hence if it has been done in the same way for other tasks, then we should leave it as is for the moment, and address this later.

Thanks for the typo catch!
Where did you notice this '--' usage?? I'll have a look so that we're being consistent and if not, to check if modifying those would break things

**Comment by bthirion** [comment]
> That was the idea, the issue comes when we have uneven number of conditions we want to contrast, for example `audio-visual`, we have 6 positive conditions for _audio_ and 10 negative for _visual_, so if I give a -1 to each of the negative conditions, I would still need to do 10/6 so still not an integer. I wonder how you've dealt with this in the past. Maybe just going to a higher factor, eg. having 5*(6 positive conditions) vs 3*(10 negative conditions)?

Yes, this is the right way to do it.

**Comment by bthirion** [comment]
LGTM.
